### PR TITLE
get rid of platform_flags.mk

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Tested-by: Jens Wiklander <jens.wiklander@linaro.org> (QEMU, FVP)
Signed-off-by: Jens Wiklander <jens.wiklander@linaro.org>

### CODE REVIEW

**Comment by jforissier** [comment]
`Tested-by: Jerome Forissier <jerome.forissier@linaro.org> (HiKey)`

`documentation/build_system.md` needs some minor update.

**Comment by jenswi-linaro** [comment]
Updated `documentation/build_system.md`

**Comment by jforissier** [comment]
Nice simplification.
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Tags applied

**Comment by jenswi-linaro** [comment]
Rebased

File: ta/ta.mk [hunk 1]
```diff
@@ -129,6 +116,12 @@ $(conf-mk-file-export): $(conf-mk-file)
 	$(q)echo CFG_ARM32_$(sm-$(@)) := $(CFG_ARM32_$(sm-$(@))) >> $@
 	$(q)echo CFG_ARM64_$(sm-$(@)) := $(CFG_ARM64_$(sm-$(@))) >> $@
 	$(q)echo CFG_TA_FLOAT_SUPPORT := $(CFG_TA_FLOAT_SUPPORT) >> $@
+	$(q)echo ta_arm32-platform-cppflags := $(ta_arm32-platform-cppflags) >> $@
```
Comment (pascal-brand-st-dev) [inline ta/ta.mk]: This is arm-specific, and we would miss some configurations for non-arm platforms.

**Comment by jenswi-linaro** [comment]
Update

**Comment by jenswi-linaro** [comment]
Update

**Comment by jforissier** [comment]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

**Comment by jenswi-linaro** [comment]
Rebased

**Pull request merged**